internal/infrastructure/kubernetes/infra_resource.go

```diff
@@ -132,6 +132,14 @@ func (i *Infra) createOrUpdateService(ctx context.Context, r ResourceRender) err
 		Name:      svc.Name,
 	}
 
+	if err := i.Client.Get(ctx, key, current); err == nil {
+		if svc.Spec.ClusterIP != current.Spec.ClusterIP && (svc.Spec.ClusterIP != "" || current.Spec.ClusterIP == "None") {
+			i.Logger.Info("recreating service to force clusterIP update", "current", current.Spec.ClusterIP, "configured", svc.Spec.ClusterIP)
+			i.Client.Delete(ctx, current)
+			current = &corev1.Service{}
+		}
+	}
```

Comment:
Is there a way with existing testing infra to implement a test which asserts success when the `clusterIP` value changes? If this is removed editing the `clusterIP` will result in errors being logged, for example:

```
2024-02-05T21:32:05.297Z    ERROR    infrastructure    runner/runner.go:70    failed to create new infra    {"runner": "infrastructure", "error": "failed to create or update service envoy-gateway-system/envoy-default-eg-e41e7b31: for Update: Service \"envoy-default-eg-e41e7b31\" is invalid: spec.clusterIPs[0]: Invalid value: []string{\"None\"}: may not change once set"}
2024-02-05T21:32:05.297Z    ERROR    watchable    message/watchutil.go:56    observed an error    {"runner": "infrastructure", "error": "failed to create or update service envoy-gateway-system/envoy-default-eg-e41e7b31: for Update: Service \"envoy-default-eg-e41e7b31\" is invalid: spec.clusterIPs[0]: Invalid value: []string{\"None\"}: may not change once set"}    
```